### PR TITLE
Add missing SMARTS strings and correctness tests for `get_fx_groups()`

### PR DESCRIPTION
**Pull request by Aariq**
Addresses #44 (partially) and #47

Adds SMARTS strings to capture all functional groups in the SIMPOL.1 paper except "number of carbons on the acid side of an amide" which I think is probably impossible with SMARTS alone.  Adds a CSV file with test compounds and their expected functional group counts as well as a test to compare to that CSV.

### CODE REVIEW

**Comment by Aariq** [comment]
Still left to figure out:

- carbon number on the acid-side of amide (tricky, maybe impossible?)
- carbonylperoxynitrate (haven't tried yet)
- nitrophenol (tricky because of definition—aromatic OH when there's also a nitro group anywhere on the ring)
- nitroesther (confused about definition)

**Comment by Aariq** [comment]
I think carbon number on the acid-side of amide is not possible with just SMARTS.  Would need to chop the bond on the acid side of the amide, then count carbons in the remaining piece and I don't think that kind of sub-setting is something SMARTS can do.

**Comment by Aariq** [comment]
I think I'll want to "Squash and Merge" this one

File: R/get_fx_groups.R [hunk 1]
```diff
@@ -78,111 +67,121 @@ get_fx_groups <- function(compound_sdf) {
...
+      ester = .data$ester - .data$carbonylperoxynitrate,
+      nitrate = .data$nitrate - .data$carbonylperoxynitrate,
+      peroxide = .data$peroxide - .data$carbonylperoxynitrate,
+      #phosphoric ester also matches phosphoric acid
+      phosphoric_acid = .data$phosphoric_acid - .data$phosphoric_ester,
+      
+      hydroxyl_aliphatic = .data$hydroxyl_total - .data$hydroxyl_aromatic, #TODO: should this happen before or after hydroyxl_aromatic is corrected? Or in the initial mutate?
+      
+      #according to SIMPOL.1 paper, nitrophenol shouldn't count aromatic hydroxyls.
+      hydroxyl_aromatic = .data$hydroxyl_aromatic - .data$nitrophenol,
```
Comment (Aariq) [inline R/get_fx_groups.R:172]: Currently tests are passing when things happen in this order, but I should double check some test cases and think about it.

**Comment by Aariq** [comment]
Still need to:
- [ ] double-check that all the functional groups have tests (i.e. the colnames in test_compounds.csv match)
- [ ] make notes on test_compounds.csv about why each compound was chosen

**Comment by Aariq** [comment]
Ready for another round of review @KristinaRiemer !

**Review by KristinaRiemer — APPROVED** [review]

**Pull request merged**